services/catalog/tests/test_endpoints.py

```diff
@@ -333,6 +330,152 @@ def test_search_with_collections_and_filter(self, client):  # pylint: disable=mi
+        elif method == "GET":
+            test_params = {"collections": "toto_S1_L1,S2_L3", "filter": "width=2500"}
+            response = client.get("/catalog/search", params=test_params)
+        assert response.status_code == fastapi.status.HTTP_404_NOT_FOUND
+
+        # Check that we get an error if at least one existing collection doesn't exist
+        if method == "POST":
+            test_json = {
+                "collections": ["S1_L1", "unexisting_collection"],
+                "filter": {
+                    "op": "and",
+                    "args": [
+                        {"op": "=", "args": [{"property": "owner"}, "toto"]},
+                        {"op": "=", "args": [{"property": "width"}, 2500]},
+                        {"op": "=", "args": [{"property": "height"}, 2500]},
+                    ],
+                },
+            }
+            response = client.post("/catalog/search", json=test_json)
+        elif method == "GET":
+            test_params = {"collections": "toto_S1_L1,unexisting_collection", "filter": "width=2500"}
+            response = client.get("/catalog/search", params=test_params)
+        assert response.status_code == fastapi.status.HTTP_404_NOT_FOUND
+
+    def test_search_using_several_collections_get(self, client):
+        """Test a search request involving several collections (GET method)"""
+        # Use implicit naming mechanism for some collections of the list + specify owner in the query parameters
+        test_params = {
+            "collections": "S1_L1,toto_S2_L3",
+            "filter-lang": "cql2-text",
+            "filter": "width=2500",
+            "owner": "toto",
+        }
+        response = client.get("/catalog/search", params=test_params)
+        assert response.status_code == 200
+        content = json.loads(response.content)
+        assert len(content["features"]) == 3
+
+        # Use implicit naming mechanism for some collections of the list + specify owner in the filter
+        test_params = {
+            "collections": "S1_L1,toto_S2_L3",
+            "filter-lang": "cql2-text",
+            "filter": "width=2500 AND owner='toto'",
+        }
+        response = client.get("/catalog/search", params=test_params)
+        assert response.status_code == 200
+        content = json.loads(response.content)
+        assert len(content["features"]) == 3
+
+        # Implicit naming mechanism will not produce the right owner_id if we don't specify it in the query
+        # parameters or in the filter
+        test_params = {"collections": "toto_S1_L1,S2_L3", "filter": "width=2500"}
+        response = client.get("/catalog/search", params=test_params)
+        assert response.status_code == fastapi.status.HTTP_404_NOT_FOUND
+
+        # Check that we get an error if at least one collection does not exist
+        test_params = {"collections": "toto_S1_L1,unexisting_collection", "filter": "width=2500"}
```

Comment:
The code first check if the original name of the collection exist in the database, if it's the case it will keep this name, otherwise the owner_id from the filter will be concatenated to the name of the collection